concordium-std/src/impls.rs

```diff
@@ -1528,6 +1528,53 @@ impl Read for ExternParameter {
     }
 }
 
+impl Seek for ExternParameter {
+    type Err = ();
+
+    fn seek(&mut self, pos: SeekFrom) -> Result<u32, Self::Err> {
+        use SeekFrom::*;
+        let end = self.size();
+        match pos {
+            Start(offset) => {
+                if offset <= end {
+                    self.current_position = offset;
+                    Ok(offset)
+                } else {
+                    Err(())
+                }
+            }
+            End(delta) => {
+                if delta > 0 {
+                    Err(()) // cannot seek beyond the end
+                } else {
+                    // due to two's complement representation of values we do not have to
+                    // distinguish on whether we go forward or backwards. Reinterpreting the bits
+                    // and adding unsigned values is the same as subtracting the
+                    // absolute value.
```

Comment:
Did I write that for the other seek implementation? It's rather an interesting choice of functionality, but I guess it saves on one instruction?